### ISSUE TITLE
New tabs/windows don't respect the `initial-command` configuration

### ISSUE THREAD

**Issue by raklaptudirm** [ghostty-org/ghostty#3329]
# Steps to reproduce:

- Add a new `initial-command` to the configuration file.
- Open Ghostty and create a new tab/window, note that the configuration isn't respected.

New window created from the initial window:
<img width="950" alt="Image" src="https://github.com/user-attachments/assets/759153a4-1fac-4d7c-89c9-8622cbe99a7b" />

New tab created in the initial window:
<img width="800" alt="Image" src="https://github.com/user-attachments/assets/1e11fed5-ff92-40d4-ae7a-2b7bdd64e954" />

The issue also seems to be present for tab splits both horizontal and vertical.

System: MacOS Sequoia 15.2

**Comment by raklaptudirm** [comment]
I seem to have misunderstood the `initial-command` option.
> This is the same as "command", but only applies to the first terminal surface created when Ghostty starts.

**Comment by chindris-mihai-alexandru** [comment]
I ran into a macOS issue while trying to use Ghostty with a tmux startup command, and it was really confusing. What threw me off was that the initial-command only applies to the first surface, and it's hard to understand how it interacts with other things like command, restored or reopened windows, and window-save-state from the documentation. 

I was expecting that if I set up a startup command, it would give me a consistent way to get into tmux every time I reopen it, but it didn't quite work that way with restored surfaces. It seemed like they didn't follow the same rules as running the configured command again. This might be how it's supposed to work, but I think the documentation could be clearer about how startup commands work with restored macOS windows.
